src/modules/Tab/Tab.js

```diff
@@ -96,19 +102,42 @@ class Tab extends Component {
     })
   }
 
+  renderColumn(columnProps) {
+    return GridColumn.create(columnProps)
+  }
+
+  renderVertical(menu) {
+    const { grid, panes } = this.props
+    const { activeIndex } = this.state
+
+    return (
+      <Grid>
+        {menu.props.tabular !== 'right' && this.renderColumn({ width: grid.tabWidth, children: menu })}
+        {this.renderColumn({
+          width: grid.segmentWidth,
+          children: _.invoke(_.get(panes, `[${activeIndex}]`), 'render', this.props),
+          stretched: true }
+        )}
+        {menu.props.tabular === 'right' && this.renderColumn({ width: grid.tabWidth, children: menu })}
+      </Grid>
+    )
+  }
+
   render() {
     const { panes } = this.props
     const { activeIndex } = this.state
 
     const menu = this.renderMenu()
+    const grid = menu.props.vertical === true ? this.renderVertical(menu) : null
     const rest = getUnhandledProps(Tab, this.props)
     const ElementType = getElementType(Tab, this.props)
 
     return (
       <ElementType {...rest}>
-        {menu.props.attached !== 'bottom' && menu}
-        {_.invoke(_.get(panes, `[${activeIndex}]`), 'render', this.props)}
-        {menu.props.attached === 'bottom' && menu}
+        {menu.props.vertical === true && grid}
```

Comment:
The `vertical` menu prop makes the case for two clean return values here.  We can also avoid creating a `null` grid in the case of non-vertical menus:

```js
if (menu.props.vertical) {
  return <ElementType {...rest}>{this.renderVertical(menu)}<ElementType>
}

// ...the old return value as-is
```